src/tls/Makefile.common

```diff
@@ -14,48 +37,82 @@ INCLUDE_PATHS= \
 	$(HACL_HOME)/code/lib \
 	$(KREMLIN_HOME)/kremlib \
 	$(MITLS_HOME)/libs/ffi \
-	./concrete-flags \
-        ./concrete-flags/$(FLAVOR)$(CONCRETE_FLAVOR)
+	$(SPECINC)
 
 ifeq (,$(MITLS_NODEBUG))
   INCLUDE_PATHS += ./concrete-flags/Debug
 else
   INCLUDE_PATHS += ./concrete-flags/NoDebug
 endif
 
-CACHE_DIR=cache.$(FLAVOR)$(CONCRETE_FLAVOR)
-
 FSTAR=$(FSTAR_HOME)/bin/fstar.exe --cache_dir $(CACHE_DIR) --cache_checked_modules $(OTHERFLAGS)
 
-EXTRACT_DIR=extract/$(FLAVOR)$(CONCRETE_FLAVOR)
+VFLAGS?=--use_hints --use_hint_hashes --record_hints --warn_error -271
+
+HINTS_DIR  ?= .
+CACHE_DIR   = cache/$(FLAVOR)$(CONCRETE_FLAVOR)
```

Comment:
is it correct to define CACHE_DIR after FSTAR, which depends on CACHE_DIR?

Reply:
I think so, but I'll move them around for clarity.